tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -305,12 +305,10 @@ class RNTupleJoinProcessor : public RNTupleProcessor {
 
 private:
    std::unique_ptr<RNTupleModel> fJoinModel;
-   /// Mapping of the auxiliary RNTuple name to its page source
-   std::unordered_map<std::string, std::unique_ptr<Internal::RPageSource>> fAuxiliaryPageSources;
+   std::vector<std::unique_ptr<Internal::RPageSource>> fAuxiliaryPageSources;
 
    std::vector<std::string> fJoinFieldNames;
-   /// Mapping of the auxiliary RNTuple name to its join index
-   std::unordered_map<std::string, std::unique_ptr<Internal::RNTupleIndex>> fJoinIndices;
+   std::vector<std::unique_ptr<Internal::RNTupleIndex>> fJoinIndices;
```

Comment:
nit: maybe some docstrings to explain that `fAuxiliaryPageSources` is indexed by ntuple, `fJoinIndices` for auxiliary ntuples (ie minus 1), but `fJoinFieldNames` applies to all ntuples?